src/panels/video-player/VideoPlayer.tsx

```diff
@@ -30,17 +30,24 @@ export const VideoPlayer = (props: VideoPlayerPropsFromParent) => {
   const checkTempVar = useCallback(
     (displayOption: string) => {
       const displayOptionVar = tempVarSyntax(displayOption);
-      const value = templateSrv.replace(displayOptionVar);
-      // Not a template var if templateSrv.replace returns the same value
+      const value = replaceVariables(displayOptionVar);
+      // Not a template var if replaceVariables returns the same value
       return value === displayOptionVar ? displayOption : value;
     },
-    [templateSrv]
+    [replaceVariables]
   );
 
   // Get display option value from the URL, or check default variable values
   const getDisplayOptionValue = useCallback(
     (queryParams: UrlQueryMap, displayOption: string) => {
-      return (queryParams[getUrlTempVarName(displayOption || '')] as string) ?? checkTempVar(displayOption);
+      const tempVarName = getUrlTempVarName(displayOption || '');
+      const tempVarVal = checkTempVar(displayOption);
+      const tempVarURLVal = queryParams[tempVarName];
+      if (Array.isArray(tempVarURLVal) || tempVarVal !== tempVarURLVal || !(tempVarURLVal as string)) {
```

Comment:
I could be misreading this, but if you're returning `tempVarVal` when `tempVarVal !== tempVarURLVal` aren't you  functionally always returning `tempVarVal`?

Reply:
You're right I need to reevaluate this

Reply:
`!(tempVarURLVal as string)` can just be `!tempVarURLVal`

Reply:
Updated to just use the saved template var value `tempVarVal` as the default, otherwise check the URL for a value.